packages/block-library/src/navigation/view.js

```diff
@@ -66,7 +66,14 @@ const { state, actions } = store(
 					actions.openMenu( 'hover' );
 			},
 			closeMenuOnHover() {
-				actions.closeMenu( 'hover' );
+				const { type, overlayOpenedBy } = getContext();
+				if (
+					type === 'submenu' &&
+					// Only open on hover if the overlay is closed.
```

Comment:
This comment was missed:

https://github.com/WordPress/gutenberg/pull/60130#discussion_r1537208503

Also a lack of braces here, which is usually error prone and might be against the coding standards too

Reply:
Good spot. It's frustrating that that code formatting issue wasn't picked up by linting. Going to raise a PR to fix this one.

Reply:
I was annoyed at the lack of braces too but I wanted to keep the same snippet that we have above for the other handler.